brvs-core/src/main/java/iroha/validation/rest/RestService.java

```diff
@@ -77,11 +80,31 @@ public Response register(@PathParam("accountId") String accountId) {
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
   public Response sendTransaction(String transaction) {
+    return sendTransaction(transaction, false);
+  }
+
+  @POST
+  @Path("/transaction/sign")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response sendTransactionWithSigning(String transaction) {
+    return sendTransaction(transaction, true);
+  }
+
+  private Response sendTransaction(String transaction, boolean sign) {
     try {
       final Builder builder = Transaction.newBuilder();
       parser.merge(transaction, builder);
-      final Transaction tx = builder.build();
-      StreamingOutput streamingOutput = output -> irohaAPI.transaction(tx)
+      // since final or effectively final is needed
+      final Transaction builtTx = builder.build();
+      final Transaction transactionsToSend;
+      if (sign) {
+        transactionsToSend = jp.co.soramitsu.iroha.java.Transaction.parseFrom(builtTx)
+            .sign(brvsAccountKeyPair).build();
+      } else {
+        transactionsToSend = builtTx;
+      }
+      StreamingOutput streamingOutput = output -> irohaAPI.transaction(transactionsToSend)
```

Comment:
What if you pass a multisig transaction? Your endpoint will be blocked until it's not committed. That might take a long time. BRVS endpoint may be stuck with long-lived multisig transactions.

Reply:
I would check that quorum is 1.

Reply:
Added a check, thanks